pandas/core/groupby.py

```diff
@@ -2368,6 +2368,13 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     # axis of the object
     if level is not None:
         if not isinstance(group_axis, MultiIndex):
+            if isinstance(level, list) or isinstance(level, range):
+                if len(level) > 1:
```

Comment:
> Only checking the length doesn't meet MI behavior.

Done by checking `len(set(level))` instead of `len(level)`.

> handle empty case.

Done by handling `len(set(level)) == 0`.

> I think it should raise if level is not scalar.

But that's the whole point of this PR and the issue it stems from!